presto-main/src/main/java/com/facebook/presto/cost/FilterStatsCalculator.java

```diff
@@ -83,6 +85,58 @@ private PlanNodeStatsEstimate filterForUnknownExpression()
         }
 
         @Override
+        protected PlanNodeStatsEstimate visitLogicalBinaryExpression(LogicalBinaryExpression node, Void context)
+        {
+            PlanNodeStatsEstimate left = process(node.getLeft());
+            PlanNodeStatsEstimate right = process(node.getRight());
+            switch (node.getType()) {
+                case AND:
+                    return intersectStats(left, right);
+                break;
+                case OR:
+                    return unionStats(left, right);
+                break;
+            }
+        }
+
+        private PlanNodeStatsEstimate unionStats(PlanNodeStatsEstimate left, PlanNodeStatsEstimate right)
+        {
+            return input; // FIXME - implement this
+        }
+
+        private PlanNodeStatsEstimate intersectStats(PlanNodeStatsEstimate left, PlanNodeStatsEstimate right)
+        {
+            PlanNodeStatsEstimate.Builder statsBuilder = PlanNodeStatsEstimate.builder();
+
+            Stream.concat(left.getSymbolsWithKnownStatistics().stream(), right.getSymbolsWithKnownStatistics().stream())
+                    .forEach(symbol -> {
+                        statsBuilder.addSymbolStatistics(symbol,
+                                intersectColumnStats(left.getSymbolStatistics(symbol), right.getSymbolStatistics(symbol)));
+                    });
+
+            double leftFilterFactor = left.getOutputRowCount() / input.getOutputRowCount();
+            double rightFilterFactor = right.getOutputRowCount() / input.getOutputRowCount();
+
+            return statsBuilder.setOutputRowCount(leftFilterFactor * rightFilterFactor * input.getOutputRowCount()).build();
```

Comment:
this does not look good.
You take into consideration the overlapping of ranges for columns in `left` and `right`stats when computing result symbol statistics. But you do not do it when you compute result `outputRowCount`